### PR TITLE
[BugFix]: Exporting when `--do_train` and `--do_eval` are not set

### PR DESCRIPTION
**Pull request by rahul-tuli**
[BugFixes]: To supported exporting when `--do_train` and `--do_eval` are not set

* Use `get_eval_dataloader()` instead of `get_val_dataloader`
* Always create a eval_dataset if num_export_samples>0

Test commands

- [X] Question answering

```bash
sparseml.transformers.train.question_answering \
--output_dir models/sparse_quantized --model_name_or_path "zoo:nlp/masked_language_modeling/bert-base/pytorch/huggingface/wikipedia_bookcorpus/12layer_pruned80_quant-none-vnni" --recipe "zoo:nlp/masked_language_modeling/bert-base/pytorch/huggingface/wikipedia_bookcorpus/12layer_pruned80_quant-none-vnni?recipe_type=transfer-question_answering" --distill_teacher "disable" --dataset_name squad --overwrite_output_dir --one_shot --num_export_samples 100
```

Relevant OutPut:
```bash
2022-06-02 15:37:39 sparseml.transformers.sparsification.trainer INFO     Exporting 100 samples to models/sparse_quantized
2022-06-02 15:37:40 sparseml.transformers.sparsification.trainer INFO     Exported 100 samples to models/sparse_quantized
[INFO|modelcard.py:460] 2022-06-02 15:37:40,391 >> Dropping the following result as it does not have all the necessary fields:
{'task': {'name': 'Question Answering', 'type': 'question-answering'}, 'dataset': {'name': 'squad', 'type': 'squad', 'args': 'plain_text'}}
```

Noting testing was done locally for all 4 scripts omitting the `--do_eval` and `--do_train` args

### CODE REVIEW

File: src/sparseml/transformers/sparsification/trainer.py [hunk 1]
```diff
@@ -502,7 +502,7 @@ def save_sample_inputs_outputs(
         device = self.model.device
 
         try:
-            dataloader = self.get_val_dataloader()
+            dataloader = self.get_eval_dataloader()
```
Comment (bfineran) [inline src/sparseml/transformers/sparsification/trainer.py:505]: good catch

**Review by bfineran — COMMENTED** [review]
lgtm overall. might be nice to create a `make_eval_dataset` bool variable or something so we don't need to repeat `training_args.do_eval or data_args.num_export_samples > 0` or make multiple changes if there's other conditions that come up in the future

File: src/sparseml/transformers/question_answering.py [hunk 2]
```diff
@@ -777,7 +777,9 @@ def compute_metrics(p: EvalPrediction):
         args=training_args,
         data_args=data_args,
         train_dataset=train_dataset if training_args.do_train else None,
-        eval_dataset=eval_dataset if training_args.do_eval else None,
+        eval_dataset=eval_dataset
+        if training_args.do_eval or data_args.num_export_samples > 0
+        else None,
```
[the remark this answers is not in the document]
Comment (rahul-tuli) [reply src/sparseml/transformers/question_answering.py:782]: Aah thanks! Addressing

File: src/sparseml/transformers/sparsification/trainer.py (hunk 1 above)
Comment (rahul-tuli) [reply src/sparseml/transformers/sparsification/trainer.py:505]: > lgtm overall. might be nice to create a `make_eval_dataset` bool variable or something so we don't need to repeat `training_args.do_eval or data_args.num_export_samples > 0` or make multiple changes if there's other conditions that come up in the future

Yay! Adding in next commit

**Review by bfineran — APPROVED** [review]

**Review by KSGulin — APPROVED** [review]

**Pull request merged**